### PR TITLE
Make SimilarityTransform act even more like RandomResizeCrop

### PR DESCRIPTION
**Pull request by satyaog**
The code of SimilarityTransform supporting a random resize crop now acts like torchvision's RandomResizeCrop

### CODE REVIEW

File: src/benzina/torch/operations.py [hunk 1]
```diff
@@ -377,11 +375,27 @@ def __call__(self, index, in_shape, out_shape, rng):
         """Return a random similarity transformation."""
         s = np.exp(rng.uniform(low=np.log(self.s[0]), high=np.log(self.s[1])))
         if self.ar is not None:
-            ar = np.exp(rng.uniform(low=np.log(self.ar[0]),
-                                    high=np.log(self.ar[1])))
-            crop_area = s * in_shape[0] * in_shape[1]
-            crop_w = np.sqrt(crop_area * ar)
-            crop_h = np.sqrt(crop_area / ar)
+            for _ in range(10):
```
Comment (satyaog) [inline src/benzina/torch/operations.py:378]: I'm not sure if we need to try 10 times before fallback to a center crop, but this is what torchvision does in [RandomResizedCrop](https://pytorch.org/docs/stable/_modules/torchvision/transforms/transforms.html#RandomResizedCrop) so maybe it's better to start with this

File: src/benzina/torch/operations.py [hunk 2]
```diff
@@ -377,11 +375,27 @@ def __call__(self, index, in_shape, out_shape, rng):
...
+                in_ar = float(in_shape[0]) / float(in_shape[1])
+                if in_ar < self.ar[0]:
+                    crop_w = in_shape[0]
+                    crop_h = int(round(crop_w / self.ar[0]))
+                elif in_ar > self.ar[1]:
+                    crop_h = in_shape[1]
+                    crop_w = int(round(crop_h * self.ar[1]))
+                else:  # whole image
+                    crop_w, crop_h = in_shape
+                self.random_crop = False
```
Comment (satyaog) [inline src/benzina/torch/operations.py:398]: The crop could still be random on one dimension instead but again, this is how torchvision does it in [RandomResizedCrop](https://pytorch.org/docs/stable/_modules/torchvision/transforms/transforms.html#RandomResizedCrop)

**Comment by satyaog** [comment]
@obilaniu , too fast ;). Still I have added some comments to read later if we want to reverted some of the changes

**Comment by obilaniu** [comment]
@satyaog Okay, I undid the merge but GitHub doesn't seem to detect I backed out the merge commit and the commit that got merged.

**Comment by satyaog** [comment]
I've recreated the PR as #27

**Pull request merged**